flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/IntermediateResultPartition.java

```diff
@@ -120,17 +118,8 @@ boolean markFinished() {
 
         hasDataProduced = true;
```

Comment:
it's better to check that this partition was not finished yet.
this can expose problems to ease trouble shooting.